libcnb-test/src/build_config.rs

```diff
@@ -251,6 +251,8 @@ pub enum BuildpackReference {
     Crate,
     /// References another buildpack by id, local directory or tarball.
     Other(String),
+
+    Local(PathBuf),
```

Comment:
@schneems we also discussed that it could be a buildpack id within your project space.

@Malax should we just rename `Other` to `Uri` and do detection in the runner for if it's a local buildpack id or path?  that way we could have it align with how values in `package.toml` are treated during packaging.

Reply:
> we also discussed that it could be a buildpack id within your project space.

Right. I like the ergonomics of that more, if it's not too difficult to implement. One thought is that it might get confusing to see:

```
Other(String::from("heroku/yarn-engine")), 
CompileLocal(String::from("heroku/nodejs-engine"))
```

Which is related to your question to Manuel.

IDK what he thinks, but my preference is generally to try to separate out refactorings from new features unless they're in conflict. It makes it easier to follow the changes and review, though it's more work for implementers. I'm suggesting we push refactoring/changing `Other()` until after this PR is merged. Also, I'm willing to do some of that work if needed.

To the end of should it be changed. I think the interface is already confusing. It was unclear to me that specifying `heroku/jvm-internal` in the java buildpack test would pull in the one from the builder and not the local one. To that end I think it's worth considering what it does already it handles:

- Path to already compiled buildpack (zipped or not)
- Buildpack Id
- Something else?

Maybe we separate these out into their own variants. For the Buildpack Id case it will work if a buildpack is specified via the builder or via the CLI flags i.e. `--buildpack <path/to/heroku/node>` I'm not sure what to name or call that. Maybe "ExternalBuildpack" or "ExternalBuildpackID"?